### DISCUSSION TITLE
Tasmota with more than 8 channels - no friendly names from channel 9 on

### DISCUSSION THREAD

**Discussion by thermseekr** [arendst/Tasmota#22310] [General]
Hello goog people,

After more than one year without touching this project, I finally resumed development and application of my ESP32/Tasmota boards. I'm deploying the 16in/16out version and I noticed that Tasmota defines friendly names for channels 1 to 8 and all channels beyond those don't show up in the Configure Other page. Also when naming the device in the Home Assistant Tasmota Integration, only channels 1 through 8 are named, channels from 9 on are kept with default names. This is not a problem at all, I just wanted to make a note of this fact in case it's something easy to address.

Below some images to illustrate the matter.

![image](https://github.com/user-attachments/assets/c8d46b1b-19b0-4089-917c-51461686b0d6)

![image](https://github.com/user-attachments/assets/50108c83-1416-473d-89b9-69b5aa49a3de)

![image](https://github.com/user-attachments/assets/ce116969-d015-4455-a2a0-4f601bf81e04)

![image](https://github.com/user-attachments/assets/3652957d-ef48-4746-b663-c55fa304f8b2)

![image](https://github.com/user-attachments/assets/379d1e86-ed64-4bc9-be7e-ac8a1950d3ef)

Warm regards,
Tales

**Comment by sfromis** [comment]
If it was "just so" to increase the number of "friendly names", it would already have happened. Has been discussed before.

**Comment by thermseekr** [comment]
Hi there. I erased my original reply as it did not add anything useful to the Community.

Well let's see. Dear @sfromis I suppose you are referring to #16768. In that case you are right, it had already been discussed. I missed that, sorry. Anyway, what I'm doing is purely reporting my findings, which now I see is nothing new. As you can see I put the topic in General conversations, not in Q&A or in Support Request. So I'm not asking for help nor even asking anything really. Hence my surprise from the bitter tone of your answer. I was not expecting that, but I will be from now on.

Checking #16768 I see that for the OP relays from 9 on would not show up neither in Config Other nor in HA. In my case relays still do not show in Config Other but they are correctly detected in HA, only they don't have a default Friendly Name. Also they work correctly without the need for any manual configuration.

Maybe the difference in the hardware is the cause for the different behavior. The board mentioned by the OP is based on 2x 74HC595, while the board I built uses 4x MCP23017, two for inputs and two for outputs, using the driver in Mode 2.

Thank you all for building and maintaining Tasmota. What you accomplished is so big and never ceases to amaze me. Very, very good job.

Regards,
Tales

**Reply by sfromis** [reply]
Well, if you picked up a "bitter tone", that was far from what I meant by a short matter-of fact post. Not my intention.

When it comes to more than 8 relays, that's another issue than "friendly names", as Tasmota can support up to 32 relay channels, but still only 8 "friendly names". It would take some restructuring of the implementation to support additional names via a file in the file system, instead of it fitting in the limited-size string pool currently used.

**Reply by thermseekr** [reply]
> Well, if you picked up a "bitter tone", that was far from what I meant by a short matter-of fact post. Not my intention.

That's the problem with written communications, the decoding of the message depends on how the receiver sees it, which in turn depends on a lot of other aspects. Forgive me for the wrong interpretation of your post.

About the friendly names, in my opinion it's not worth to even touch that. It's not a problem, channels > 8 work perfectly even if names are not automatically assigned to them, and also the percentage of users that deploy hardware with more that 8ch must be really small.

Have a nice weekend, thank you for taking the time to reply.

Tales
